src/Features/CSharp/Portable/CodeFixes/AddExplicitCast/AddExplicitCastCodeFixProvider.cs

```diff
@@ -0,0 +1,399 @@
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.AddExplicitCast), Shared]
+    [ExtensionOrder(After = PredefinedCodeFixProviderNames.ChangeToYield, Before = PredefinedCodeFixProviderNames.GenerateConstructor)]
+    internal partial class AddExplicitCastCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        /// <summary>
+        /// CS0266: Cannot implicitly convert from type 'x' to 'y'. An explicit conversion exists (are you missing a cast?)
+        /// </summary>
+        private const string CS0266 = nameof(CS0266);
+
+        /// <summary>
+        /// CS1503: Argument 1: cannot convert from 'double' to 'int'
+        /// </summary>
+        private const string CS1503 = nameof(CS1503);
+        private const int MaximumConversionOptions = 3;
+
+        [ImportingConstructor]
+        public AddExplicitCastCodeFixProvider()
+        {
+        }
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds
+        {
+            get => ImmutableArray.Create(CS0266, CS1503);
+        }
+
+        internal sealed override CodeFixCategory CodeFixCategory => CodeFixCategory.CodeStyle;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+            var diagnostic = context.Diagnostics.First();
+
+            var root = await document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            if (root != null && syntaxFacts != null && semanticModel != null &&
+                TryGetTargetNode(root, diagnostic.Location.SourceSpan) is ExpressionSyntax targetNode)
+            {
+                var exactSolution = GetTypeInfo(semanticModel, root, targetNode, cancellationToken, out var nodeType, out var conversionType, out var potentialConvTypes);
+                if (exactSolution)
+                {
+                    context.RegisterCodeFix(new MyCodeAction(
+                        CSharpFeaturesResources.Add_explicit_cast,
+                        c => FixAsync(context.Document, context.Diagnostics.First(), c)),
+                        context.Diagnostics);
+                }
+                else if (potentialConvTypes.Length > 1)
+                {
+                    var actions = new ArrayBuilder<CodeAction>();
+                    for (var i = 0; i < Math.Min(MaximumConversionOptions, potentialConvTypes.Length); i++)
+                    {
+                        var convType = potentialConvTypes[i];
+                        actions.Add(new MyCodeAction(string.Format(CSharpFeaturesResources.Convert_type_to_0, convType.ToDisplayString()),
```

Comment:
`ToDisplayString` can be a very long string if it generates fully qualified name. Maybe better to just use `convType.Name`?

Reply:
I used `convType.Name` before, but looks like it doesn't work for some cases. For example, it shows `Func` instead of `System.Func<Derived, Derived>`, I'll try to see if there is a better solution for this.